c_src/quicer_stream.c

```diff
@@ -426,6 +426,191 @@ async_accept_stream2(ErlNifEnv *env,
+  ERL_NIF_TERM res = ATOM_ERROR_INTERNAL_ERROR;
+  QuicerConnCTX *c_ctx = NULL;
+  ERL_NIF_TERM active_val;
+  ERL_NIF_TERM eopen_flag;
+  ERL_NIF_TERM estart_flag;
+  unsigned int open_flag = QUIC_STREAM_OPEN_FLAG_NONE; // default
+  uint32_t sendflags = 0;
+
+  if (4 != argc)
+    {
+      return ERROR_TUPLE_2(ATOM_BADARG);
+    }
+
+  ERL_NIF_TERM eHandle = argv[0];
+  ERL_NIF_TERM ebin = argv[1];
+  ERL_NIF_TERM eoptions = argv[2];
+  ERL_NIF_TERM eFlags = argv[3];
+
+  if (!enif_get_resource(env, eHandle, ctx_connection_t, (void **)&c_ctx))
+    {
+      return ERROR_TUPLE_2(ATOM_BADARG);
+    }
+
+  if (!enif_get_map_value(
+          env, eoptions, ATOM_QUIC_STREAM_OPTS_ACTIVE, &active_val))
+    {
+      return ERROR_TUPLE_2(ATOM_BADARG);
+    }
+
+  // optional open_flag,
+  if (enif_get_map_value(
+          env, eoptions, ATOM_QUIC_STREAM_OPTS_OPEN_FLAG, &eopen_flag))
+    {
+      if (!enif_get_uint(env, eopen_flag, &open_flag))
+        {
+          // if set must be valid.
+          return ERROR_TUPLE_2(ATOM_BADARG);
+        }
+      // @TODO set event mask for some flags
+    }
+
+  if (enif_get_map_value(
+          env, eoptions, ATOM_QUIC_STREAM_OPTS_START_FLAG, &estart_flag))
+    {
+      // We do not allow start flag here.
+      return ERROR_TUPLE_2(ATOM_BADARG);
+    }
+
+  //
+  // note, s_ctx is not shared yet, thus no locking is needed.
+  //
+  QuicerStreamCTX *s_ctx = init_s_ctx();
+
+  if (!s_ctx)
+    {
+      return ERROR_TUPLE_2(ATOM_ERROR_NOT_ENOUGH_MEMORY);
+    }
+
+  // This is optional
+  get_uint32_from_map(env, eoptions, ATOM_QUIC_EVENT_MASK, &s_ctx->event_mask);
+
+  enif_keep_resource(c_ctx);
+  s_ctx->c_ctx = c_ctx;
+
+  // Caller should be the owner of this stream.
+  s_ctx->owner = AcceptorAlloc();
+
+  if (!s_ctx->owner)
+    {
+      res = ERROR_TUPLE_2(ATOM_ERROR_NOT_ENOUGH_MEMORY);
+      goto ErrorExit;
+    }
+
+  if (!enif_self(env, &(s_ctx->owner->Pid)))
+    {
+      res = ERROR_TUPLE_2(ATOM_BAD_PID);
+      goto ErrorExit;
+    }
+
+  if (!set_owner_recv_mode(s_ctx->owner, env, active_val))
+    {
+      res = ERROR_TUPLE_2(ATOM_BADARG);
+      goto ErrorExit;
+    }
+
+  if (QUIC_FAILED(Status = MsQuic->StreamOpen(c_ctx->Connection,
+                                              open_flag,
+                                              ClientStreamCallback,
+                                              s_ctx,
+                                              &(s_ctx->Stream))))
+    {
+
+      res = ERROR_TUPLE_3(ATOM_STREAM_OPEN_ERROR, ATOM_STATUS(Status));
+      goto ErrorExit;
+    }
+
+  // Now we have Stream handle
+  s_ctx->eHandle = enif_make_resource(s_ctx->imm_env, s_ctx);
+
+  QuicerStreamSendCTX *send_ctx = init_send_ctx();
+  if (!send_ctx)
+    {
+      return ERROR_TUPLE_2(ATOM_ERROR_NOT_ENOUGH_MEMORY);
+    }
+
+  ErlNifBinary *bin = &send_ctx->bin;
+
+  if (enif_get_uint(env, eFlags, &sendflags))
+    {
+      enif_self(env, &send_ctx->caller);
```

Comment:
newbie question: doesn't the return value of this need to be checked like [this line above](https://github.com/emqx/quic/pull/160/files#diff-eaeae12e4b178768da074307e3d45ccee17502c30573712225fd0d9f7d5e4b1aR506)?

Reply:
because it is already checked in that line.